src/main/java/org/spongepowered/api/world/WorldSaveMode.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.world;
+
+/**
+ * The enumeration that contains all save modes for saving worlds.
+ */
+public enum WorldSaveMode {
+    /**
+     * The save mode that marks all world chunks and metadata (depending on
+     * {@link SerializationBehavior}) to save in the future. The mode does not
+     * guarantee one-time save, nor flush.
+     */
+    SAVE,
```

Comment:
I think this is fine as-is. Most I/O operations have the pattern that they eventually flush unless you explicitly call to do so. I don't think its worth it to make the name more explicit on how the save is performed.

Reply:
I'm speaking about in general how I/O operations are actually implemented in Java or in other languages as an API. When you open a handle and try write to it, it doesn't actually do anything with it yet but buffers it. You can write more and you might not see anything on disk. But eventually, something might show up unless the program crashes in the middle and you may explicitly call flush and those changes are immediately available.

So in my opinion this is something I expect how these types of APIs work. You do the operation and optionally flush for immediate results.

Reply:
Ah, I see. Looks like I misunderstood the last part. I can agree on that the SAVE option could be misleading and developers could now expect it to mean save and flush. Maybe we could make the enum names more explicit? `SAVE_WORLD`, `SAVE_METADATA`, `SAVE_CHUNKS`. Not really my favorite but it would not be as ambiguous?

Or perhaps the best way is to actually just introduce new method, `saveAndFlush`?